module/VuFind/src/VuFind/View/Helper/Root/Content.php

```diff
@@ -0,0 +1,101 @@
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Mario Trojan <mario.trojan@uni-tuebingen.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+use Laminas\View\Helper\AbstractHelper;
+use VuFind\Content\PageLocator;
+
+/**
+ * Content View Helper to resolve translated pages.
+ * This is basically a wrapper around the PageLocator.
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Mario Trojan <mario.trojan@uni-tuebingen.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Content extends AbstractHelper
+{
+    /**
+     * PageLocator instance to resolve translated pages.
+     *
+     * @var PageLocator
+     */
+    protected $pageLocator;
+
+    /**
+     * Context View Helper instance to resolve translated pages.
+     *
+     * @var Context
+     */
+    protected $contextHelper;
+
+    /**
+     * Constructor
+     *
+     * @param PageLocator $pageLocator   Page locator
+     * @param Context     $contextHelper Context view helper
+     */
+    public function __construct(PageLocator $pageLocator,
+        Context $contextHelper
+    ) {
+        $this->pageLocator = $pageLocator;
+        $this->contextHelper = $contextHelper;
+    }
+
+    /**
+     * Search for a translated template and render it using a temporary context.
+     *
+     * @param string $pageName    Name of the page
+     * @param string $pathPrefix  Path where the template should be located
+     * @param array  $context     Optional array of context variables
+     * @param array  $pageDetails Optional output variable for additional info
+     * @param string $pattern     Optional file system pattern to search page
+     *
+     * @return string            Rendered template output
+     */
+    public function renderTranslated($pageName, $pathPrefix='content',
+        $context=[], &$pageDetails=[], $pattern=null
+    ) {
+        if (!str_ends_with($pathPrefix, '/')) {
+            $pathPrefix .= '/';
+        }
+        $pathPrefix = 'templates/' . $pathPrefix;
+        $pageDetails = $this->pageLocator->determineTemplateAndRenderer(
+            $pathPrefix, $pageName, $pattern
+        );
+        $path = str_replace(
```

Comment:
How about making `findContainingTheme()` return a relativePath element as well, so we could avoid this rather ugly replace?

Reply:
This has been changed now; however, findContainingTheme is used not only for templates, so i used str_replace anyway to strip the `templates/` prefix.

Reply:
I think an unconstrained `str_replace` might have issues here in the unlikely but possible situation where the string occurs more than once in the path; maybe it's better to `preg_replace` `|^templates/|` or use a `substr` to ensure greater precision.